### PR TITLE
add tests for reduction with pointwise epilogue

### PR DESCRIPTION
**Pull request by liqiangxl**
First step of #2063 
Added tests for:
(1) inner reduction + non-broadcast epilogue
(2) outer reduction + non-broadcast epilogue
(3) inner reduction + broadcast epilogue
(4) outer reduction + broadcast epilogue

### CODE REVIEW

File: tests/cpp/test_reduction_pointwise.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      {t_ref},
+      __LINE__,
+      __FILE__);
+};
+} // namespace
+TEST_F(PointwiseFusedReductionTest, InnerReductionNonBroadcast) {
+  const int reduction_dim = 1;
+  ReductionNonBroadcast(reduction_dim);
+}
+TEST_F(PointwiseFusedReductionTest, OuterReductionNonBroadcast) {
```
Comment (jjsjann123) [inline tests/cpp/test_reduction_pointwise.cpp:65]: so even though we are adding all four tests, the `OuterReductionNonBroadcast` is really the only one that the entire fusion goes through a single kernel. The other three would just segment into `reduction` + `pointwise`?

**Comment by jjsjann123** [comment]
One more question here is that, this is only for single arith operation behind a reduction, so we are not yet considering use this to verify the other plan we have that uses `# of fusion ops to cap unroll_factor`?

File: tests/cpp/test_reduction_pointwise.cpp (hunk 1 above)
Comment (liqiangxl) [reply tests/cpp/test_reduction_pointwise.cpp:65]: You are right. I believe all other three cases are segmented due to performance reason and we need to revisit this decision by optimizing the generated code and add corresponding benchmark cases in the followed PRs.

**Comment by liqiangxl** [comment]
> One more question here is that, this is only for single arith operation behind a reduction, so we are not yet considering use this to verify the other plan we have that uses `# of fusion ops to cap unroll_factor`?

Not yet. This work is checking how to schedule reduction with epilogue ops. The heuristic revision primarily focuses on prelogue ops. Need to check the influence of epilogue ops on optimal heuristics after the prelogue ops are investigated. Then, combine them, looks even more difficult to have one works for all!

**Review by jjsjann123 — APPROVED** [review]
Thanks for adding the test.

**Pull request merged**